flask_security/decorators.py

```diff
@@ -53,7 +53,11 @@ def _get_unauthorized_view():
             except BuildError:
                 view = None
         utils.do_flash(*utils.get_message('UNAUTHORIZED'))
-        return redirect(view or request.referrer or '/')
+        redirect_to = '/'
+        if request.referrer and not request.referrer.endswith(request.path):
```

Comment:
If taking `path.full_path` and removing the query string is the same as `request.path`, of course it won't change the meaning of the tests...